src/de.hpi.swa.graal.squeak/src/de/hpi/swa/graal/squeak/nodes/ExecuteContextNode.java

```diff
@@ -141,7 +144,7 @@ public final Object executeResumeInMiddle(final VirtualFrame frame, final long i
     private GetOrCreateContextNode getGetOrCreateContextNode() {
         if (getOrCreateContextNode == null) {
             CompilerDirectives.transferToInterpreterAndInvalidate();
-            getOrCreateContextNode = insert(GetOrCreateContextNode.create(code));
+            getOrCreateContextNode = insert(GetOrCreateContextNode.create(code, false));
```

Comment:
This seems to be the only time, `fromActiveProcess` is set to `false`. Is this correct? If yes, why did you have to set it to `false` here? The node is used to trigger a `ProcessSwitch` to itself. This way, we protect GraalSqueak from a `StackOverflowError`.

Reply:
On the contrary, the stack overflow seems to be the rare exception. The node is actually used in the process switch branch as well as the non virtual return, precisely the places where we switch process to some other process. I thought the rarely used stack overflow check was not worth treating specially